store/tikv/gc_worker.go

```diff
@@ -0,0 +1,332 @@
+	if err != nil {
+		w.done <- errors.Trace(err)
+	}
+	err = w.doGC(safePoint)
+	if err != nil {
+		w.done <- errors.Trace(err)
+	}
+	w.done <- nil
+}
+
+func (w *GCWorker) resolveLocks(safePoint uint64) error {
+	req := &kvrpcpb.Request{
+		Type: kvrpcpb.MessageType_CmdScanLock.Enum(),
+		CmdScanLockReq: &kvrpcpb.CmdScanLockRequest{
+			MaxVersion: proto.Uint64(safePoint),
+		},
+	}
+	bo := NewBackoffer(gcResolveLockMaxBackoff)
+
+	log.Infof("[gc worker] %s start resolve locks, safePoint: %v.", w.uuid, safePoint)
+	startTime := time.Now()
+	regions, totalResolvedLocks := 0, 0
+
+	var key []byte
+	for {
+		select {
+		case <-w.quit:
+			return errors.New("[gc worker] gc job canceled")
+		default:
+		}
+
+		region, err := w.store.regionCache.GetRegion(bo, key)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		resp, err := w.store.SendKVReq(bo, req, region.VerID())
+		if err != nil {
+			return errors.Trace(err)
+		}
+		if regionErr := resp.GetRegionError(); regionErr != nil {
+			err = bo.Backoff(boRegionMiss, errors.New(regionErr.String()))
+			if err != nil {
+				return errors.Trace(err)
+			}
+			continue
+		}
+		locksResp := resp.GetCmdScanLockResp()
+		if locksResp == nil {
+			return errors.Trace(errBodyMissing)
+		}
+		if locksResp.GetError() != nil {
+			return errors.Errorf("unexpected scanlock error: %s", locksResp)
+		}
+		locksInfo := locksResp.GetLocks()
+		locks := make([]*Lock, len(locksInfo))
+		for i := range locksInfo {
+			locks[i] = newLock(locksInfo[i])
+		}
+		ok, err1 := w.store.lockResolver.ResolveLocks(bo, locks)
+		if err1 != nil {
+			return errors.Trace(err)
+		}
+		if !ok {
+			err = bo.Backoff(boTxnLock, err1)
+			if err != nil {
+				return errors.Trace(err)
+			}
+			continue
+		}
+		regions++
+		totalResolvedLocks += len(locks)
+		key = region.EndKey()
+		if len(key) == 0 {
+			break
+		}
+	}
+	log.Infof("[gc worker] %s finish resolve locks, safePoint: %v, regions: %v, total resolved: %v, cost time: %s", w.uuid, safePoint, regions, totalResolvedLocks, time.Now().Sub(startTime))
+	return nil
+}
+
+func (w *GCWorker) doGC(safePoint uint64) error {
+	req := &kvrpcpb.Request{
+		Type: kvrpcpb.MessageType_CmdGC.Enum(),
+		CmdGcReq: &kvrpcpb.CmdGCRequest{
+			SafePoint: proto.Uint64(safePoint),
+		},
+	}
+	bo := NewBackoffer(gcMaxBackoff)
+
+	log.Infof("[gc worker] %s start gc, safePoint: %v.", w.uuid, safePoint)
+	startTime := time.Now()
+	regions := 0
+
+	var key []byte
+	for {
+		select {
+		case <-w.quit:
+			return errors.New("[gc worker] gc job canceled")
+		default:
+		}
+
+		region, err := w.store.regionCache.GetRegion(bo, key)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		resp, err := w.store.SendKVReq(bo, req, region.VerID())
+		if err != nil {
+			return errors.Trace(err)
+		}
+		if regionErr := resp.GetRegionError(); regionErr != nil {
+			err = bo.Backoff(boRegionMiss, errors.New(regionErr.String()))
+			if err != nil {
+				return errors.Trace(err)
+			}
+			continue
+		}
```

Comment:
Line137-157 and line207-227 code are look like duplicated. Can they make a common method?